ci/kind/kind-setup.sh

```diff
@@ -0,0 +1,333 @@
+    fi
+  done
+
+  # create new bridge network per subnet
+  i=0
+  networks=()
+  for s in $SUBNETS; do
+    network=$CLUSTER_NAME-$i
+    echo "creating network $network with $s"
+    docker network create -d bridge --subnet $s $network >/dev/null 2>&1
+    networks+=($network)
+    i=$((i+1))
+  done
+
+  num_networks=${#networks[@]}
+  if [[ $num_networks -eq 0 ]]; then
+    networks+=("bridge")
+    num_networks=$((num_networks+1))
+  fi
+
+  i=0
+  for node in $nodes; do
+    network=${networks[i]}
+    docker network connect $network $node >/dev/null 2>&1
+    echo "connected worker $node to network $network"
+    node_ip=$(docker inspect $node --format '{{range $i, $conf:=.NetworkSettings.Networks}}{{$conf.IPAddress}}{{end}}')
+
+    # reset network
+    docker exec -t $node ip link set eth1 down
+    docker exec -t $node ip link set eth1 name eth0
+    docker exec -t $node ip link set eth0 up
+    gateway=$(echo "${node_ip/%?/1}")
+    docker exec -t $node ip route add default via $gateway
+    echo "node $node is ready with ip change to $node_ip with gw $gateway"
+
+    # change kubelet config before reset network
+    docker exec -t $node sed -i "s/node-ip=.*/node-ip=$node_ip/g" /var/lib/kubelet/kubeadm-flags.env
+    docker exec -t $node pkill kubelet
+    docker exec -t $node pkill kube-proxy
+    i=$((i+1))
+    if [[ $i -ge $num_networks ]]; then
+      i=0
+    fi
+  done
+
+  for node in $nodes; do
+    node_ip=$(docker inspect $node --format '{{range $i, $conf:=.NetworkSettings.Networks}}{{$conf.IPAddress}}{{end}}')
+    while true; do
+      tmp_ip=$(kubectl describe node $node | grep InternalIP)
+      if [[ "$tmp_ip" == *"$node_ip"* ]]; then
+        break
+      fi
+      echo "current ip $tmp_ip, wait for new node ip $node_ip"
+      sleep 2
+    done
+  done
+}
+
+function delete_networks {
+  networks=$(docker network ls -f name=$CLUSTER_NAME --format '{{.Name}}')
+  networks="$(echo $networks)"
+  if [[ ! -z $networks ]]; then
+    docker network rm $networks > /dev/null 2>&1
+    echo "deleted networks $networks"
+  fi
+}
+
+function load_images {
+  echo "load images"
+  set +e
+  for img in $IMAGES; do
+    docker image inspect $img > /dev/null 2>&1
+    if [[ $? -ne 0 ]]; then
+      echoerr "docker image $img not found"
+      continue
+    fi
+    kind load docker-image $img --name $CLUSTER_NAME > /dev/null 2>&1
+    if [[ $? -ne 0 ]]; then
+      echoerr "docker image $img failed to load"
+      continue
+    fi
+    echo "loaded image $img"
+  done
+  set -e
+}
+
+function create {
+  if [[ -z $CLUSTER_NAME ]]; then
+    echoerr "cluster-name not provided"
+    exit 1
+  fi
+
+  set +e
+  kind get clusters | grep $CLUSTER_NAME > /dev/null 2>&1
+  if [[ $? -eq 0 ]]; then
+    echoerr "cluster $CLUSTER_NAME already created"
+    exit 0
+  fi
+  set -e
+
+  config_file="/tmp/kind.yml"
+  cat <<EOF > $config_file
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  disableDefaultCNI: true
```

Comment:
we can do that. But I choose not to install any CNI at all if ANTREA_CNI is false. Because I do not know the consequence of install Antrea on top Kubelet default CNI, and wants to keep it safe.